Framework/PythonInterface/mantid/api/src/Exports/IFunction.cpp

```diff
@@ -35,6 +36,32 @@ PyObject *getCategories(IFunction &self) {
   return registered;
 }
 
+/// An object for constructing a shared_ptr that won't ever delete its pointee
+class NoDeleting {
+public:
+  /// Does nothing
+  void operator()(void *) {}
+  /// Does nothing
+  void operator()(const void *) {}
+};
+
+//------------------------------------------------------------------------------------------------------
+/**
+* Something that makes IFunction return to python a composite function
+* for Product function, Convolution or
+* any similar superclass of composite function.
+* @param self :: Enables it to be called as a member function on the
+* IFunction class
+*/
+Mantid::API::CompositeFunction_sptr castToCompositeFunction(IFunction &self) {
```

Comment:
I see what you mean. For a `CompositeFunctionWrapper` object, accessing its component functions and their associated parameters through operator `.` is not possible since most of the identifier strings (e.g. "Gaussian 0", "f0.f1", or "f0.f1.Height) are not valid python identifiers. Then, the only pythonic access interface is through `__getitem__` and `__settitem__`. I guess the choice is to either use  `.` for parameters and attributes owned by the function, or dispose of this functionality.